rholang/src/main/scala/coop/rchain/rholang/interpreter/Substitute.scala

```diff
@@ -141,9 +133,12 @@ object Substitute {
               news = news,
               matches = matches,
               ids = term.ids,
-              locallyFree = locallyFree,
-              connectiveUsed = connectiveUsed
+              locallyFree = term.locallyFree.until(env.shift),
+              connectiveUsed = term.connectiveUsed
             )
+          sortedPar <- ParSortMatcher.sortMatch[M](par)
```

Comment:
This may be a minor thing, but sorting this par causes all of the insides to be sorted. Maybe we want to split substitute so that there's an external API that sorts and an internal API that doesn't

I don't want you to add anything to this PR, just something to think about.